src/components/Card.js

```diff
@@ -1,53 +1,70 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { isLiked, name, link, _id },
+    cardSelector,
+    handleImageClick,
+    handleDeleteButton,
+    handleLikeButtonClick,
+    removeLikeButton
+  ) {
+    this._isLiked = isLiked;
     this._name = name;
     this._link = link;
+    this._id = _id;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteButton = handleDeleteButton;
+    this._handleLikeButtonClick = handleLikeButtonClick;
+    this._removeLikeButton = removeLikeButton;
   }
 
   _setEventListeners() {
     this._cardElement
       .querySelector(".card__like-button")
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.

```js
  generateCard() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```